Simplenote/Classes/UITextView+Simplenote.swift

```diff
@@ -28,6 +28,76 @@ extension UITextView {
 }
 
 
+// MARK: - Updating!
+//
+extension UITextView {
+
+    /// Inserts the specified Text at a given range.
+    /// - Note: Resulting selected range will end up at the right hand side of the newly inserted text.
+    ///
+    func insertText(text: String, in range: Range<String.Index>) {
+        registerUndoCheckpointAndPerform { storage in
+            let range = self.text.utf16NSRange(from: range)
+            storage.replaceCharacters(in: range, with: text)
+            self.selectedRange = NSRange(location: range.location + text.count, length: .zero)
```

Comment:
I believe it is incorrect to add `text.count` to NSRange. `text.count` is based on the representation of the string (utf8 / utf16) and can result in a wrong adjustment.

I would try to do:
```
let insertedTextRange = NSRange(text.fullRange, in: text)
self.selectedRange = NSRange(location: range.location + insertedTextRange.length, length: .zero)
```

Reply:
You're 💯 on spot. I need to fix mac as well (this code was borrowed from there!!)